### PR TITLE
Trilio DataMover API charm

### PR DESCRIPTION
**Pull request by abhijeetpatra**
- It follows openstack-api charm model.
- Template file has been added to generate dmapi.conf

### CODE REVIEW

File: juju-charms/charm-trilio-data-mover-api/src/files/trilio/trilio-data-mover-api.service [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[Unit]
+Description=Datamover API service
+
+[Service]
+User = root
```
Comment (MuralidharB) [inline juju-charms/charm-trilio-data-mover-api/src/files/trilio/trilio-data-mover-api.service]: This need to be run as nova or dmapi user.

File: juju-charms/charm-trilio-data-mover-api/src/lib/charm/openstack/dmapi.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    service_name = name = 'dmapi'
+
+    # First release supported
+    release = 'queens'
+
+    # Packages the service needs installed
+    packages = ['python-nova']
+
+    # Init services the charm manages
+    services = ['trilio-data-mover-api']
```
Comment (MuralidharB) [inline juju-charms/charm-trilio-data-mover-api/src/lib/charm/openstack/dmapi.py]: can we keep this name consistent. I saw @maheshbhoge using as 'tvault-datamover-api'. Please check with him

File: juju-charms/charm-trilio-data-mover-api/src/files/trilio/trilio-data-mover-api.service (hunk 1 above)
Comment (abhijeetpatra) [reply juju-charms/charm-trilio-data-mover-api/src/files/trilio/trilio-data-mover-api.service]: Ok, I will add dmapi user then.

File: juju-charms/charm-trilio-data-mover-api/src/files/trilio/tvault-datamover-api.service [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[Unit]
+Description=Datamover API service
+
+[Service]
+User = dmapi
```
Comment (abhijeetpatra) [inline juju-charms/charm-trilio-data-mover-api/src/files/trilio/tvault-datamover-api.service]: added dmapi user

File: juju-charms/charm-trilio-data-mover-api/src/lib/charm/openstack/dmapi.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    service_name = name = 'dmapi'
+
+    # First release supported
+    release = 'queens'
+
+    # Packages the service needs installed
+    packages = ['python-nova']
+
+    # Init services the charm manages
+    services = ['tvault-datamover-api']
```
Comment (abhijeetpatra) [inline juju-charms/charm-trilio-data-mover-api/src/lib/charm/openstack/dmapi.py]: changed the service name to 'tvault-datamover-api'

**Pull request merged**